src/components/mapper/edit-ast.ts

```diff
@@ -0,0 +1,191 @@
+ * @return {File} New ast
+ */
+export const AddSuccessFailureSteps = 
+    (ast: File, currentStep: string, nextStep: string, stepType:string) : File => {
+        const path = GetPathOfStep(ast, currentStep);
+        const successPath = GetSuccessFailurePath(path.node, path.scope, path.parentPath, path.state, stepType);
+        if (successPath === null) {
+            const key = (stepType==="success")?syntax.onSuccess : syntax.onFail;
+            if(path.node.arguments.length===1) {
+                path.node.arguments.push(createObjectExpressionWithProperty(key, nextStep));
+            }else{
+                path.node.arguments[1].properties.push(createSuccessFailurePropertyWithStep(key, nextStep));
+            }
+        } else {
+            successPath[0].value.body.body.push(createExpressionStatement(nextStep));
+        }
+        return ast;
+    };
+
+/**
+ * Add a new step as a the success or failure step before a given step.
+ *
+ * @param {File} ast
+ * @param {string} beforeStep - which step the new step is going to be add before
+ *
+ * @return {File} New ast
+ */
+export const AddSuccessFailureStepsBefore = (ast: File, beforeStep:string) : File => {
+    const pathBefore = GetPathOfStep(ast, beforeStep);
+    traverse(ast, {
+        CallExpression(path: any) {
+            if (path.node.callee.name === syntax.stepExecutor && path.node.arguments[0].value > +beforeStep-1) {
+                path.node.arguments[0].value = path.node.arguments[0].value + 1;
+            }
+        }
+    });
+    const args = pathBefore.parentPath.parent.body[pathBefore.parentPath.key];
+    pathBefore.parentPath.parent.body[pathBefore.parentPath.key] = createExpressionStatementWithSuccess(
+        beforeStep, args
+    );
+    return ast;
+};
+
+/**
+ * Add a new step as a the success or failure step before a given condition.
+ *
+ * @param {File} ast
+ * @param {string} condition - which condition the new step is going to be add before
+ *
+ * @return {[File, number]} New ast, New step id
+ */
+export const AddSuccessFailureStepsBeforeCondition = (ast: File, condition:string) : [File, number] => {
+    const [lastStep, pathBefore] = GetConditionPathWithLastStep(ast, condition);
+    traverse(ast, {
+        CallExpression(path: any) {
+            if (path.node.callee.name === syntax.stepExecutor && path.node.arguments[0].value > lastStep) {
+                path.node.arguments[0].value = path.node.arguments[0].value + 1;
+            }
+        }
+    });
+    const args = pathBefore.parent.body[pathBefore.key];
+    pathBefore.parent.body[pathBefore.key] = createExpressionStatementWithSuccess(lastStep + 1, args);
+    return [ast, lastStep+1];
+};
+
+/**
+ * Add a condition to the ast after a given step with given params.
+ *
+ * @param {File} ast
+ * @param {string} step
+ * @param {string} condition
+ * @param {any} params
+ *
+ * @return {File} New ast
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
Do we have to suppress the rule due to the usage of `File`? Let's figure out a way to whitelist this type in eslint config. We can fix this in the next PR. 

cc:/ @HimashiRathnayake

Reply:
Fixed with #12